addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/stop-ec2-environment/stop-ec2-environment-sc.js

```diff
@@ -0,0 +1,149 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const _ = require('lodash');
+const StepBase = require('@aws-ee/base-workflow-core/lib/workflow/helpers/step-base');
+
+class StopEc2EnvironmentSc extends StepBase {
+  async start() {
+    const environmentId = await this.payload.string('environmentId');
+    this.state.setKey('STATE_ENVIRONMENT_ID', environmentId);
+
+    const requestContext = await this.payload.object('requestContext');
+    this.state.setKey('STATE_REQUEST_CONTEXT', requestContext);
+
+    const instanceId = await this.payload.string('instanceIdentifier');
+
+    const ec2 = await this.getEc2Service();
+
+    const params = {
+      InstanceIds: [instanceId],
+    };
+
+    let instanceStatusInfo;
+    try {
+      const paramsForDescribe = { ...params, IncludeAllInstances: true };
+      instanceStatusInfo = await ec2.describeInstanceStatus(paramsForDescribe).promise();
```

Comment:
What is the behavior of describe API if the instanceId doesn't exist?

Reply:
It would throw and invalid-id exception when the instanceId does not exist.